### ISSUE TITLE
ERROR: <urllib3.connection.HTTPSConnection object at 0x7ff24b791100>: Failed to establish a new connection:  [Errno -2] Name or service not known

### ISSUE THREAD

**Issue by anilsakr** [Azure/azure-cli#27315]
### Describe the bug

I'm encountering a connectivity issue when trying to run the following Azure CLI command in our QA environment:
`az storage blob upload-batch -s ./ -d '$web' --account-name "$(STORAGE_ACCOUNT_NAME)" --overwrite --account-key "$(STORAGE_ACCOUNT_KEY)"`
Problem
When running this command in the QA environment, I'm receiving the following error message:
`ERROR: <urllib3.connection.HTTPSConnection object at 0x7ff24b791100>: Failed to establish a new connection: [Errno -2] Name or service not known`

### Related command

az storage blob upload-batch -s ./ -d '$web' --account-name "$(STORAGE_ACCOUNT_NAME)" --overwrite --account-key "$(STORAGE_ACCOUNT_KEY)"

### Errors

ERROR: <urllib3.connection.HTTPSConnection object at 0x7ff24b791100>: Failed to establish a new connection: [Errno -2] Name or service not known

### Issue script & Debug output

ERROR: <urllib3.connection.HTTPSConnection object at 0x7ff24b791100>: Failed to establish a new connection: [Errno -2] Name or service not known

### Expected behavior

It should upload the files to Azure storage account

### Environment Summary

Task         : Azure CLI
Description  : Run Azure CLI commands against an Azure subscription in a PowerShell Core/Shell script when running on Linux agent or PowerShell/PowerShell Core/Batch script when running on Windows agent.
Version      : 2.225.0
Author       : Microsoft Corporation

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by anilsakr** [comment]
Also wanted to mention, Actually same pipeline code and network configs are working for Dev environment but when I run in QA its throwing this error.
As mentioned in another issue tried nslookup login.microsoftonline.com in self hosted agent, its able to resolve, below is the out put,
Server:		10.60.12.8
Address:	10.60.12.8#53

Non-authoritative answer:
login.microsoftonline.com	canonical name = login.mso.msidentity.com.
login.mso.msidentity.com	canonical name = ak.privatelink.msidentity.com.
ak.privatelink.msidentity.com	canonical name = www.tm.ak.prd.aadg.trafficmanager.net.
Name:	www.tm.ak.prd.aadg.trafficmanager.net
Address: 20.190.190.129
Name:	www.tm.ak.prd.aadg.trafficmanager.net
Address: 20.190.190.130
Name:	www.tm.ak.prd.aadg.trafficmanager.net
Address: 20.190.190.131
Name:	www.tm.ak.prd.aadg.trafficmanager.net
Address: 20.190.190.132

**Comment by calvinhzy** [comment]
What is the cli version by running `az version`

**Comment by PeterThomasAwen** [comment]
Just wanted to say I also get a similar error if the account name was incorrect, 
when using az storage file download-batch  
I'd do an echo and check (STORAGE_ACCOUNT_NAME) is what you expect it to be.

Maybe a better error can be provided?

Running Az version 2.52.0

**Comment by fabio-s-franco** [comment]
> Just wanted to say I also get a similar error if the account name was incorrect, when using az storage file download-batch I'd do an echo and check (STORAGE_ACCOUNT_NAME) is what you expect it to be.
> 
> Maybe a better error can be provided?
> 
> Running Az version 2.52.0

Agreed, I had a similar experience with the key vault, the message is not helpful to pinpoint the problem, in fact it will mislead towards unrelated issues.

**Comment by calvinhzy** [comment]
Hi @sofia-valles, can you help provide the detailed environment info? Are you also using a QA environment and what OS/terminal are you using? Wondering if any of the solutions in https://github.com/Azure/azure-cli/issues/21247 might be helpful.

**Comment by r4hulp** [comment]
> Is there any update on this issue? I'm also getting this error upon running az storage blob upload. I've already confirmed that I am referring to the correct storage account, permissions are in place and the resolving works.

Did it work for you? This is failing in our case as well. We are trying to upload a blob via Azure DevOps Pipeline.

**Comment by r4hulp** [comment]
@calvinhzy - In our case, it is one of the Azure DevOps agent.

```
Starting: Upload key tab to repository
==============================================================================
Task         : Azure CLI
Description  : Run Azure CLI commands against an Azure subscription in a PowerShell Core/Shell script when running on Linux agent or PowerShell/PowerShell Core/Batch script when running on Windows agent.
Version      : 2.230.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/deploy/azure-cli
==============================================================================
/usr/bin/az --version
azure-cli                         2.57.0

core                              2.57.0
telemetry                          1.1.0

Extensions:
azure-devops                      0.26.0

Dependencies:
msal                              1.26.0
azure-mgmt-resource             23.1.0b2
```

**Comment by almmechanics** [comment]
Likewise - i'm experiencing exactly the same with  Azure CLI `2.57.0` on build agent `ubuntu-latest` on `3.234.0`

**Comment by Itsamirmasoud** [comment]
Experiencing the same thing when trying to batch upload to an Azure Storage account. We have 4 env, works well on 3 and fails on 1 of them.

**Comment by yusbel** [comment]
I'm receiving the same error when listing secrets from Azure KeyVault or setting new secrets. I receive this error using the portal shell and using locally the az cli after sucessfull authentication.

 az keyvault secret list --vault-name "keyvaultname" -o table
<urllib3.connection.HTTPSConnection object at 0x7f2cad608d60>: Failed to establish a new connection: [Errno -2] Name or service not known

Any hint? Thank you

**Comment by yusbel** [comment]
I was able to connect without error after restarting the computer

**Comment by kasireb** [comment]
I have same error message, trying to list az keyvault secrets/keys from az cmdline.

**Comment by mayban29** [comment]
Getting same error connecting to keyvault

<urllib3.connection.HTTPSConnection object at 0x7fa54aa94580>: Failed to establish a new connection: [Errno -2] Name or service not known

i am using azure cli

**Comment by Hetzeilen** [comment]
The cli expects plain resource names, not quoted.   What was failing on me:
`az storage container create --name  'mon<redacted>hdget' --account-name 'p<redacted>sa' --auth-mode login`

What succeeded:
`az storage container create --name  'mon<redacted>hdget' --account-name p<redacted>sa --auth-mode login`

**Comment by PravallikaKonchada** [comment]
ERROR: <urllib3.connection.HTTPSConnection object at 0x7b175f65a1b0>: Failed to resolve '----.vault.azure.net' ([Errno -2] Name or service not known)

I'm using WSL to get the secrets from Keyvault and resulting in above error
